internal/controller/storageclient_controller.go

```diff
@@ -424,6 +424,18 @@ func (r *storageClientReconcile) reconcilePhases() (ctrl.Result, error) {
 	if utils.AddAnnotation(&r.storageClient, utils.DesiredConfigHashAnnotationKey, storageClientResponse.DesiredStateHash) {
 		update = true
 	}
+
+	if storageClientResponse.RbdDriverRequirements != nil && len(storageClientResponse.RbdDriverRequirements.TopologyDomainLables) > 0 {
+		annotationValue := strings.Join(storageClientResponse.RbdDriverRequirements.TopologyDomainLables, ",")
+		if utils.AddAnnotation(&r.storageClient, utils.TopologyDomainLabelsAnnotationKey, annotationValue) {
+			update = true
+		}
+	} else {
+		// remove the annotation if it exists
+		delete(r.storageClient.GetAnnotations(), utils.TopologyDomainLabelsAnnotationKey)
+		update = true
```

Comment:
There is a subtle bug here: this code will create an update even if nothing got removed (an update on every reconcile)
```suggestion
		annotations  : = r.storageClient.GetAnnotations()
		annotationCount :=  len(annotations)
		delete(annotations, utils.TopologyDomainLabelsAnnotationKey)
		update = len(annotations) < annotationCount 
```

Reply:
Okay I will have a Util function